timesketch/frontend-ng/src/components/Studio/SigmaRuleModification.vue

```diff
@@ -0,0 +1,257 @@
+      <pre>{{ editingRule.search_query }}</pre>
+      <!-- TODO: This should become a global class how to nicely highlight a query -->
+    </div>
+  </v-container>
+</template>
+  
+<script>
+import ApiClient from '../../utils/RestApiClient'
+import { SigmaTemplates, defaultSigmaPlaceholder } from '@/utils/SigmaRuleTemplates'
+import EventBus from '../../main'
+import _ from 'lodash'
+
+export default {
+  props: ['rule_uuid', 'sigmaRule'],
+  data() {
+    return {
+      editingRule: { ruleYaml: defaultSigmaPlaceholder }, // empty state
+      status_chip_text: 'OK',
+      status_text: '',
+      ruleYamlTextArea: {},
+      SigmaTemplates: SigmaTemplates,
+      search: '',
+      isNewRule: false,
+      isUpdatingRule: false,
+      isParsingSuccesful: false,
+    }
+  },
+  watch: {
+    rule_uuid: function (newVal, oldVal) {
+      if (newVal === 'new') {
+        this.editingRule = {
+          title: 'New Sigma Rule',
+        }
+        this.isNewRule = true
+        this.isUpdatingRule = false
+        this.status_chip_text = 'Ok'
+        this.isParsingSuccesful = true
+        this.ruleYamlTextArea = defaultSigmaPlaceholder
+        this.editingRule.rule_yaml = defaultSigmaPlaceholder
+        this.parseSigma(this.editingRule.rule_yaml)
+      } else {
+        this.getRuleByUUID(newVal)
+      }
+    },
+  },
+  loaded() {
+    this.parseSigma(this.editingRule.rule_yaml)
+  },
+  mounted() {
+    if (this.rule_uuid === 'new') {
+      this.editingRule = {
+        title: 'New Sigma Rule',
+      }
+      this.isNewRule = true
+      this.isUpdatingRule = false
+      this.status_chip_text = 'Ok'
+      this.isParsingSuccesful = true
+      this.ruleYamlTextArea = defaultSigmaPlaceholder
+      this.editingRule.rule_yaml = defaultSigmaPlaceholder
+      this.parseSigma(this.editingRule.rule_yaml)
+    } else {
+      this.getRuleByUUID(this.rule_uuid)
+    }
+  },
+  methods: {
+    selectTemplate(text) {
+      var matchingTemplate = this.SigmaTemplates.find((obj) => {
+        return obj.title === text
+      })
+      this.ruleYamlTextArea = matchingTemplate.text
+      this.parseSigma(matchingTemplate.text)
+    },
+    // Set debounce to 300ms if parseSigma is used.
+    parseSigma: _.debounce(function (ruleYaml) {
+      // eslint-disable-line
+      ApiClient.getSigmaRuleByText(ruleYaml)
+        .then((response) => {
+          var parsedRule = response.data.objects[0]
+          if (!parsedRule.author) {
+            this.status_text = 'No Author given'
+            this.isParsingSuccesful = false
+          } else {
+            this.editingRule = parsedRule
+            this.isParsingSuccesful = true
+            this.status_chip_text = 'Ok'
+            this.status_text = ''
+          }
+        })
+        .catch((e) => {
+          this.status_text = e.response.data.message
+          this.isParsingSuccesful = false
+          this.status_chip_text = 'ERROR'
+        })
+    }, 300),
+    statusColors() {
```

Comment:
Ack for your previous comment, but then let's make this a property and not a method.

Reply:
actually I changed my mind, lets make it inline.